cost-analyzer/values-agent.yaml

```diff
@@ -44,9 +44,9 @@ prometheus:
       runAsUser: 1001
     extraSecretMounts:
       - name: object-store-volume
-        mountPath: /etc/thanos/config 
+        mountPath: /etc/thanos/config
         readOnly: true
-        secretName: {{ .Values.agentKeySecretName }}
+        secretName: kubecost-agent-object-store
```

Comment:
Same question. Do we not want to set this to a values option?

Reply:
can you get this command to work? I can't figure it out. Seems like you can't template in a values file?

```sh
helm template cost-analyzer --repo https://kubecost.github.io/cost-analyzer/ \
 -f https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-agent.yaml \
 --set agentKeySecretName=myname
```

```
Error: failed to parse https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-agent.yaml: error converting YAML to JSON: yaml: invalid map key: map[interface {}]interface {}{".Values.agentKeySecretName":interface {}(nil)}
```

Reply:
This is the command we share with customers, that is broken with the last agent change PR (which, I think is due to templating in the `values-agent.yaml`):

```
helm install kubecost kubecost/cost-analyzer \
  --namespace kubecost-agent \
--values=https://raw.githubusercontent.com/kubecost/cost-analyzer-helm-chart/develop/cost-analyzer/values-agent.yaml \
  --set prometheus.server.global.external_labels.cluster_id=test \
  --set kubecostProductConfigs.clusterName=test \
  --set-file agentKey="kubecost-agent.key"
```

with my PR, this works again:
```
helm install kubecost kubecost/cost-analyzer \
  --namespace kubecost-agent \
  -f cost-analyzer/values-agent.yaml \
  --set prometheus.server.global.external_labels.cluster_id=test \
  --set kubecostProductConfigs.clusterName=test \
  --set-file agentKey="kubecost-agent.key" \
  --create-namespace
```
With this PR, the secret name could be changed, you would just uncomment that line and edit the line for the thanos-sidecar secret.

Reply:
I am totally on board with this change, I was just double checking to see if we could somehow require less replication of settings across a values and command line. I still think that's possible, but you guys are working with the customers the most, and if you aren't running into problematic scenarios with the current process, then I fully support it.

Reply:
Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required. 

Will merge after approval.

Reply:
> Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required.
> 
> Will merge after approval.

Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required.

Will merge after approval.

Reply:
> I am totally on board with this change, I was just double checking to see if we could somehow require less replication of settings across a values and command line. I still think that's possible, but you guys are working with the customers the most, and if you aren't running into problematic scenarios with the current process, then I fully support it.

Okay, the PR fixes an issue, and I don't want to wait too long for conflicts. We can address the secret creation when it is required.

Will merge after approval.